lib/android-helpers.js

```diff
@@ -474,8 +475,17 @@ helpers.unlockWithHelperApp = async function (adb) {
     flags: "0x10200000",
     stopApp: false
   };
-  await adb.startApp(startOpts);
-  await adb.startApp(startOpts);
+  await retryInterval(10, 3000, async () => {
+    if (!(await adb.isScreenLocked())) {
+      return;
+    }
+    await adb.forceStop(UNLOCK_HELPER_PKG_ID);
+
+    logger.info("Unlocking screen");
+
+    // The command takes too much time so we should not call the command over twice in the same method.
```

Comment:
this implementation might call `startApp` up to 10 times in the worst case though

Reply:
If you'd like to limit startApp duration then it is possible either to play with its arguments (https://github.com/appium/appium-adb/blob/c49cb460839a7e5e3265812e54a2b02fe231460d/lib/tools/apk-utils.js#L69) or you could set forced promise timeout with bluebird (http://bluebirdjs.com/docs/api/timeout.html)